angularSTEP/src/app/item-dialog/item-dialog.component.ts

```diff
@@ -0,0 +1,38 @@
+import {Component, Inject} from '@angular/core';
+import {FormBuilder, FormControl, FormGroup, Validators} from '@angular/forms';
+import {MAT_DIALOG_DATA, MatDialogRef} from '@angular/material/dialog';
+
+@Component({
+  selector: 'app-item-dialog',
+  templateUrl: './item-dialog.component.html',
+  styleUrls: ['./item-dialog.component.scss']
+})
+export class ItemDialogComponent {
+  form: FormGroup = new FormGroup({
+    quantity: new FormControl(1, Validators.required),
+  })
+  selectedItem: string;
+  add: boolean;
+  remove: boolean;
+  max: number;
+
+  constructor(
+    private fb: FormBuilder,
+    private dialogRef: MatDialogRef<ItemDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) data: any,
+  ) { 
+    this.selectedItem = data.selectedItem;
+    this.add =  data.add;
+    this.remove = data.remove;
+    this.max = data.max;
+  }
+
+  close() {
+    this.dialogRef.close(this.form.controls.quantity.value);
```

Comment:
I don't see anywhere assigning to this.dialogRef, but is that because it's a special magic value that is declared+assigned "for free"?

Reply:
Yep, I believe that gets injected by magic.

Reply:
I was looking into this more to see if I could give you a clearer answer besides "special magic value" but as far as I can tell it is just kind of the magic of how the dialog is created and used